src/Analyser/MutatingScope.php

```diff
@@ -5807,6 +5810,36 @@ public function getConstantReflection(Type $typeWithConstant, string $constantNa
 		return $typeWithConstant->getConstant($constantName);
 	}
 
+	/** @return array<string, ExpressionTypeHolder> */
+	private function getSuperglobalTypes(): array
+	{
+		$superglobalTypes = [];
+		$exprStrings = ['$GLOBALS', '$_SERVER', '$_GET', '$_POST', '$_FILES', '$_COOKIE', '$_SESSION', '$_REQUEST', '$_ENV'];
+		foreach ($this->expressionTypes as $exprString => $typeHolder) {
+			if (!in_array($exprString, $exprStrings, true)) {
+				continue;
+			}
+
+			$superglobalTypes[$exprString] = $typeHolder;
+		}
+		return $superglobalTypes;
+	}
+
+	/** @return array<string, ExpressionTypeHolder> */
+	private function getNativeSuperglobalTypes(): array
```

Comment:
you could de-duplicate this method by passing in the expressionTypes via parameter

Reply:
right. I stole this basically from `getNativeConstantTypes()` vs `getConstantTypes()` which could be adapted in a similar way then. maybe in a follow-up refactor :)